lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/impl/util/InitialDataImportUtil.java

```diff
@@ -0,0 +1,98 @@
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.eclipse.jdt.annotation.NonNull;
+import org.opendaylight.mdsal.common.api.LogicalDatastoreType;
+import org.opendaylight.mdsal.dom.api.DOMDataBroker;
+import org.opendaylight.mdsal.dom.api.DOMDataTreeWriteTransaction;
+import org.opendaylight.yangtools.yang.data.api.YangInstanceIdentifier;
+import org.opendaylight.yangtools.yang.data.api.schema.DataContainerChild;
+import org.opendaylight.yangtools.yang.data.api.schema.NormalizedNode;
+import org.opendaylight.yangtools.yang.data.impl.schema.builder.impl.ImmutableContainerNodeBuilder;
+import org.opendaylight.yangtools.yang.data.util.DataSchemaContextTree;
+import org.opendaylight.yangtools.yang.model.api.EffectiveModelContext;
+import org.opendaylight.yangtools.yang.model.api.SchemaContext;
+import org.opendaylight.yangtools.yang.model.api.SchemaNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class InitialDataImportUtil {
+    private static final Logger LOG = LoggerFactory.getLogger(InitialDataImportUtil.class);
+    public static final long IMPORT_TIMEOUT_MILLIS = 20_000;
+
+    private InitialDataImportUtil() {
+        throw new UnsupportedOperationException("Init of utility class is forbidden");
+    }
+
+    private static void importConfigDatastoreFromJSON(InputStream inputStream,
+                                                      SchemaContext schemaContext,
+                                                      EffectiveModelContext effectiveModelContext,
+                                                      DOMDataBroker dataBroker)
+            throws IOException, InterruptedException, ExecutionException, TimeoutException, SerializationException {
+        LOG.info("Loading data into config datastore from JSON");
+        SchemaNode rootSchemaNode = DataSchemaContextTree.from(effectiveModelContext).getRoot().getDataSchemaNode();
+        JsonNodeConverter jsonNodeConverter = new JsonNodeConverter(schemaContext);
+        try (Reader reader =
+                     new InputStreamReader(inputStream, Charset.defaultCharset())) {
+            NormalizedNode<?, ?> nodes = jsonNodeConverter.deserialize(rootSchemaNode, reader);
+            // For some reason JsonParserStream.parse() doesn't wrap deserialized NormalizedNode in root schema node
+            // (urn:ietf:params:xml:ns:netconf:base:1.0)data as XMLParserStream does..
+            // Wrap it here:
+            nodes = ImmutableContainerNodeBuilder.create()
+                    .withNodeIdentifier(new YangInstanceIdentifier.NodeIdentifier(
+                            rootSchemaNode.getQName()))
+                    .addChild((DataContainerChild<? extends YangInstanceIdentifier.PathArgument, ?>) nodes)
+                    .build();
+            DOMDataTreeWriteTransaction wrTrx = dataBroker.newWriteOnlyTransaction();
```

Comment:
Extract writing to datastore to separate method as it is identical in both json and xml cases. Or create method to convert InputStream to NormalizedNode according to file format.
Anyway, remove this duplicity.